src/mpi/datatype/typerep/src/typerep_yaksa_create.c

```diff
@@ -49,6 +49,19 @@ int MPIR_Typerep_create_vector(int count, int blocklength, int stride, MPI_Datat
     MPIR_ERR_CHKANDJUMP(rc, mpi_errno, MPI_ERR_INTERN, "**yaksa");
 
     YAKSA_UPDATE_NEWTYPE(newtype);
+    if (HANDLE_IS_BUILTIN(oldtype)) {
+        MPI_Aint el_sz = (MPI_Aint) MPIR_Datatype_get_basic_size(oldtype);
+        newtype->n_builtin_elements = count * blocklength;
+        newtype->builtin_element_size = el_sz;
+        newtype->basic_type = oldtype;
+    } else {
+        MPIR_Datatype *old_dtp;
+        MPIR_Datatype_get_ptr(oldtype, old_dtp);
+
+        newtype->n_builtin_elements = count * blocklength * old_dtp->n_builtin_elements;
+        newtype->builtin_element_size = old_dtp->builtin_element_size;
+        newtype->basic_type = old_dtp->basic_type;
+    }
```

Comment:
Shouldn't this go into the newly created `YAKSA_UPDATE_NEWTYPE` function/macro?

Reply:
This is the thing with function vs macro. Once we make it into a function, it becomes an independent compilation entity and it needs better, more complete, and less flexible abstraction. For example, once we move the above-mentioned code into the function, the function will host code from all different datatypes into a general context. In another view, we are pulling code from their own context where is narrow and simple into a general context where it requires an understanding of all datatype contexts. Function, in my view, incurs more accidental complexity than macros.

But here, I think we can manage. So I will make the change.